classes/form/CustomerAddressForm.php

```diff
@@ -217,4 +209,82 @@ function (FormField $item) {
             'formFields' => $formFields,
         ];
     }
+
+    /**
+     * Performs validation on field values.
+     * Returns true if all field values are correct, false otherwise.
+     *
+     * @return bool
+     */
+    private function validateFieldsValues(): bool
+    {
+        $isValid = true;
+
+        $isValid &= $this->validatePostcode();
+        $isValid &= $this->validateField('firstname', 'isName', $this->translator->trans(
+            'Invalid name',
+            [],
+            'Shop.Forms.Errors'
+        ));
+        $isValid &= $this->validateField('lastname', 'isName', $this->translator->trans(
+            'Invalid name',
+            [],
+            'Shop.Forms.Errors'
+        ));
+        $isValid &= $this->validateField('city', 'isCityName', $this->translator->trans(
+            'Invalid format.',
+            [],
+            'Shop.Forms.Errors'
+        ));
+
+        return (bool) $isValid;
+    }
+
+    /**
+     * @return bool
+     */
+    private function validatePostcode(): bool
+    {
+        $postcode = $this->getField('postcode');
+        if ($postcode && $postcode->isRequired()) {
+            $country = $this->formatter->getCountry();
+            if (!$country->checkZipCode($postcode->getValue())) {
+                $postcode->addError($this->translator->trans(
+                    'Invalid postcode - should look like "%zipcode%"',
+                    ['%zipcode%' => $country->zip_code_format],
+                    'Shop.Forms.Errors'
+                ));
+
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param string $fieldName
+     * @param string $validationFunction
+     * @param string $validationFailMessage
+     *
+     * @return bool
+     */
+    private function validateField(string $fieldName, string $validationFunction, string $validationFailMessage): bool
+    {
+        $field = $this->getField($fieldName);
+        if (null === $field) {
+            return true;
+        }
+        $value = $field->getValue();
+        if ($field->isRequired() && empty($value)) {
+            return false;
+        }
+        if (!empty($value) && false === (bool) Validate::$validationFunction($value)) {
+            $field->AddError($validationFailMessage);
```

Comment:
```suggestion
            $field->addError($validationFailMessage);
```
Just notice the method name is not the right one.